compose/automation/expr_types.go

```diff
@@ -171,26 +174,61 @@ func CastToComposeRecordValues(val interface{}) (out types.RecordValueSet, err e
 	switch val := expr.UntypedValue(val).(type) {
 	case *types.Record:
 		return val.Values, nil
+	case *types.RecordValue:
+		return types.RecordValueSet{val}, nil
 	case types.RecordValueSet:
 		return val, nil
 	case map[string]string:
-		out = types.RecordValueSet{}
-		for k, v := range val {
-			out = out.Set(&types.RecordValue{Name: k, Value: v})
+		for _, k := range slice.Keys(val) {
+			out = out.Set(&types.RecordValue{Name: k, Value: val[k]})
 		}
-
 		return
-
 	case map[string][]string:
-		out = types.RecordValueSet{}
-		for k, vv := range val {
-			for i, v := range vv {
+		for _, k := range slice.Keys(val) {
+			for i, v := range val[k] {
 				out = out.Set(&types.RecordValue{Name: k, Value: v, Place: uint(i)})
 			}
 		}
-
 		return
+	case map[string]interface{}:
+		for _, k := range slice.Keys(val) {
+			if val[k] == nil {
+				continue
+			}
```

Comment:
`nil` can be a bit tricky when working with `interface{}`.
I would prefer to see something in the lines of https://github.com/cortezaproject/corteza-server/blob/2021.6.x-fix-wf-copy-record/pkg/expr/func_generic.go#L42 with some additional tests to cover this specific execution path.
You can copy-paste that function into the `compose/automation` package.

Ref: https://yourbasic.org/golang/gotcha-why-nil-error-not-equal-nil/